bindings/cpp/include/wtf/macros.h

```diff
@@ -44,6 +45,19 @@
 #define WTF_THREAD_ENABLE(name) \
   WTF_THREAD_ENABLE_IF(kWtfEnabledForNamespace, name)
 
+// Enables WTF tracing for any thread based on whether the current namespace
+// is enabled. Thread name will be auto-generated.
+#define WTF_ALL_THREADS_ENABLE() \
```

Comment:
Good idea - if you do it this way, it will be better supported across platforms and a bit cheaper in the already-enabled case.

// Enables the current thread based on a condition.
// Does nothing if the thread has already been enabled. Thread name will be auto-generated.
#define WTF_AUTO_THREAD_IF(condition) \
  if (condition && \
      !__INTERNAL_WTF_NAMESPACE::PlatformGetThreadLocalEventBuffer()) { \
    __INTERNAL_WTF_NAMESPACE::Runtime::GetInstance()->EnableCurrentThread( \
        std::hash<std::thread::id>{}(std::this_thread::get_id()), nullptr, __FILE__);   \
  }

// Same as WTF_AUTO_THREAD_IF(condition) where the condition is whether
// the current namespace is enabled.
#define WTF_AUTO_THREAD() \
  WTF_AUTO_THREAD_IF(condition)